processing/src/main/java/org/apache/druid/query/filter/Filter.java

```diff
@@ -31,12 +31,27 @@
 import org.apache.druid.segment.vector.VectorColumnSelectorFactory;
 
 import javax.annotation.Nullable;
+import java.util.LinkedHashSet;
 import java.util.Map;
 import java.util.Set;
 
 @SubclassesMustOverrideEqualsAndHashCode
 public interface Filter
 {
+  default String getFilterString()
+  {
+    return toString();
+  }
+
+  /**
+   * Returns a LinkedHashSet of all child filters for this filter with no duplicates.
+   * <p>The ordering of child filters is important in some cases, e.x.short-curcuiting.</p>
+   */
+  default LinkedHashSet<Filter> getFilters()
+  {
+    return new LinkedHashSet<>();
+  }
```

Comment:
true - i'm planning to make it return an `ImmutableSortedSet` which returns a static empty set just by `ImmutableSortedSet.of()`. Also it would indicate that child filters are not modifiable once constructed. would add more javadoc indicating only AndFilter/OrFilter have child filters. The `BooleanFilter` class is not a good name IMO, boolean would make sense for every type of filter?

Reply:
actually ended up using `ImmutableSet` since filter is not comparable, and `ImmutableSet` is ordered so that seems enough.